src_c/surface.c

```diff
@@ -1879,6 +1890,77 @@ surf_blit(pgSurfaceObject *self, PyObject *args, PyObject *keywds)
     return pgRect_New(&dest_rect);
 }
 
+static PyObject *
+surf_rblit(pgSurfaceObject *self, PyObject *const *args, Py_ssize_t nargs)
+{
+    SDL_Surface *src, *dest = pgSurface_AsSurface(self);
+    SDL_Rect dest_rect, unused_rect;
+    SDL_Rect *src_rect;
+    PyObject *argpos;
+    PyObject *srcobject;
+    int result, flags_numeric = 0;
+
+    if (nargs == 0 || nargs > 3)
+        return RAISE(
+            PyExc_ValueError,
+            "Function requires at least 2 positional arguments in the order: "
+            "source, dest, (special_flags optional)");
+    else if (nargs == 3)
+        if (!pg_IntFromObj(args[2], &flags_numeric))
```

Comment:
pg_IntFromObj also takes floats. You want PyLong_AsLong. See the documentation for the error checking requirements for that, it's more specific than most.